azurerm/internal/services/eventgrid/tests/eventgrid_system_topic_resource_test.go

```diff
@@ -0,0 +1,222 @@
+package tests
+
+import (
+	"fmt"
+	"net/http"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/acceptance"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func TestAccAzureRMEventGridSystemTopic_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_eventgrid_system_topic", "test")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMEventGridSystemTopicDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMEventGridSystemTopic_basic(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMEventGridSystemTopicExists(data.ResourceName),
+					resource.TestCheckResourceAttrSet(data.ResourceName, "source"),
+					resource.TestCheckResourceAttrSet(data.ResourceName, "topic_type"),
+					resource.TestCheckResourceAttrSet(data.ResourceName, "metric_resource_id"),
+				),
+			},
+			data.ImportStep(),
+		},
+	})
+}
+
+func TestAccAzureRMEventGridSystemTopic_requiresImport(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_eventgrid_system_topic", "test")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMEventGridSystemTopicDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMEventGridSystemTopic_basic(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMEventGridSystemTopicExists(data.ResourceName),
+				),
+			},
+			{
+				Config:      testAccAzureRMEventGridSystemTopic_requiresImport(data),
+				ExpectError: acceptance.RequiresImportError("azurerm_eventgrid_system_topic"),
+			},
+		},
+	})
+}
+
+func TestAccAzureRMEventGridSystemTopic_basicWithTags(t *testing.T) {
```

Comment:
This can be 
```suggestion
func TestAccAzureRMEventGridSystemTopic_complete(t *testing.T) {
```

instead of with tags